server/daemon/debian/kanidmd.service

```diff
@@ -10,13 +10,16 @@ Before=radiusd.service
 [Service]
 Type=notify
 DynamicUser=yes
-StateDirectory=kanidm
+User=kanidmd_dyn
+Group=kanidmd_dyn
```

Comment:
What's the effective owner/group of things like the backups folder and its contents when the daemon is running with this config? I'd be concerned that it makes it impossible to get to backups without adding the backup user to the kanidmd_dyn group (which would be impossible). Probably makes more sense to keep the user as kanidmd and systemd will limit its access outside the given folders with its controls

Reply:
Yeaaah, I didn't consider the backup scenario and I don't think there's a clean way to apply the `kanidmd` group to them.

The next best I can think of is trying to stay within DynamicUser, but using the `kanidmd` group as the effective group instead of supplemental. That would automatically infer group privileges to a known GID. But need to test if DynamicUser allows a static group without losing the dynamic user. If that fails, then we replicate the implied controls of DynamicUser but without the actual dynamic UID/GID.

Reply:
It _seems_ to work but I'm not sure how exactly. What the systemd NSS module reports doesn't fully match reality, i.e. the dynamic user definitely is getting the permissions of the static group, but from the perspective of NSS it is not a member of the group, and it's basegroup is still created matching the value given to User. So this might be undefined behavior that while seemingly beneficial, may turn out later to be a bad idea.

Reply:
Yep, that seems like the best of all possible options